src/components/singleArticle.jsx

```diff
@@ -28,20 +29,50 @@ function SingleArticle() {
 
   if (isError) return <p>❌ Error Fetching Information</p>;
 
+  const onClick = (votes) => {
+    setUserVote((userVote) => {
+      setIsError(null);
+      return userVote + votes;
+    });
+    patchArticle(article_id, votes).catch((err) => {
+      setIsError("❌ Something went wrong, please try again");
+    });
+  };
+
   return (
     <section className="tile" key={article.article_id}>
       <img src={article.article_img_url} alt="article img"></img>
       <h1>{article.title}</h1>
       <p className="date">Posted: {article.created_at.substring(0, 10)}</p>
       <p className="author">Author: {article.author}</p>
       <p>{article.body}</p>
-      <p className="votes">Votes: {article.votes}</p>
+      {isError ? <p>{isError}</p> : null}
+      <button
+        type="button"
+        onClick={() => {
+          onClick(1);
+        }}
+        disabled={userVote !== 0}
+      >
+        ❤️ {article.votes + userVote}
+      </button>
+      <button
+        type="button"
+        onClick={() => {
+          onClick(-1);
+        }}
+        disabled={userVote !== 0}
+      >
+        👎
+      </button>
```

Comment:
Really like the use of emojis! The vote button is great, its persistent and the user can only click once, good stuff 😄